### DISCUSSION TITLE
Trouble with DietPi and Amiberry quitting AmigaOS to Linux login prompt

### DISCUSSION THREAD

**Discussion by n1b5** [BlitterStudio/amiberry#979] [Q&A]
I've been fiddling about and installing various OSes (Amiga) on my Pi4 inside my lovely Checkmate case.  All seems really lovely - apart from one thing I can't figure out and it's a bit of a deal-breaker; so I'm hoping someone here may know the answer.

The issue is it keeps dropping back to the linux command prompt login screen, seemingly randomly, killing emulation and I don't know why.  Often it will kill UAE when copying large files from my PC0(linux) drive to the AmigaOS, but also if I leave a game running long enough, it does the same thing - no warning, just straight to the linux prompt.

Here's my config:
rpi4.  4gig.  Amiberry on DietPi 64bit. 
3.2 classic and manually set up in winuae.  
I have tried 040, 030 and 020 with 128/8 or 4 chip, rtg installed and large m/sd (64gig - 32 for linux, the rest divided 3 ways for Amiga), but for now:

68040 JIT, FPU internal, cpu - fastest (tried 25mhz too and no JIT).  Not set up MMU yet.
AGA, wait for blitter, no fast copper, A1200 chipset, multithreaded draw, sprites and S vs Playfield
3.2 ROM with Original UAE expansion
8 chip, 128 Z3 
I x 64gb m/sd card - 32 for DietPi, 32 for AmigaOS32 (2 gig Workbench, 2 x c15gig partitions, PC0 volume pointing to Dietpi/home/....etc
Built in expansions
UAE zorro III/32mb with chipset refresh
1024x768 fullscreen on 19inch LCD and Amiga screen 640x512, HighRes normal, aspect ratio and remove artifacts
Sound: emulated A1200 with buffer 1 at 'pull audio'
Port 0 system mouse
Misc:  Faster RT, Untrap - middle button, mouse capture, retroarch quit and menu button, serial port RTS/CTS

Sorry for the long post, but I'm keen to make this work.

**Comment by solskogen** [comment]
Does this happen on other host OS like Arch Linux / Manjaro or Raspberry Pi OS? 
Also, can you enable logging and post the relevant lines from amiberry.log when the crash happens? Are you using HDFs? If yes, does it happen without HDFs?

**Reply by n1b5** [reply]
Hi and thanks.  I've never had the issue using AmiKit, RPi on the Pi.  I'll look to the logging in Linux and see what it reveals.   Could it possibly be the Linux kernel is doing something automated that is stealing focus from Amiberry, do you think?
I'm using and .hdf, partitioned 3 ways inside AmigaOS.

**Reply by midwan** [reply]
I don't think this is a matter of "stealing focus", otherwise Amiberry would still be running in the background. I think it's probably crashing out for some reason (would be interesting to know why, the logs might help with that).

Also, which version of Amiberry are you using? SDL2 or Dispmanx?

**Reply by n1b5** [reply]
No, good point, of course.  I'm trying to replicate the issue now, but not had it since.  Perhaps I shouldn't complain at that?  My version is the Dispmanx.  I'm seeing nothing of value from any log file inside /var/log.  Is there somewhere else I should be looking?  Thanks
